src/Http/Client/FormDataPart.php

```diff
@@ -219,6 +228,28 @@ public function __toString()
 
         return $out;
     }
+
+    /**
+     * Get the string for the header parameter.
+     *
+     * If the value contains non-ASCII letters an additional header indicating
+     * the charset encoding will be set.
+     *
+     * @param string $name The name of the header parameter
+     * @param string $value The value of the header parameter
+     * @return string
+     */
+    protected function _headerParameterToString($name, $value)
+    {
+        // This regex matches non-ASCII characters and double quotes
+        $pattern = '/[^\x20-\x21|\x23-\x7E]/';
+        $return = sprintf('%s="%s"', $name, preg_replace($pattern, '', $value));
+        if (preg_match($pattern, $value) && $this->_charset != null) {
```

Comment:
Use `!== null` as suggested by Scrutinizer, also move the null check first as it's cheaper than preg match.